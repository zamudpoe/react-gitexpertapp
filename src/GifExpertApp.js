import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

import "animate.css"

const GifExpertApp = ({ defaultCategory = [] }) => {
  /* const [ categories, setCategories ] = useState( ["One Punch"] ) */
  const [ categories, setCategories ] = useState( defaultCategory ) 

  return (
    <>
      <h2 className="animate__animated animate__bounce animate__delay_2s">GifExpertApp</h2>
      <hr />

      {/* Le pasamos la funcion seteadora de estado "setCategories" por medio de la prop del mismo nombre */}
      <AddCategory setCategories = { setCategories } />

      <ol>       
        { 
          categories.map( ( category, idx ) => (
            <GifGrid 
              key      = { idx } 
              category = { category } 
            />
          ))

        } 
      </ol> 
    </>
  )
} 

/* 
GifExpertApp.propTypes = { 
  defaultCategory: PropTypes.string.isRequired
}
*/

export default GifExpertApp


