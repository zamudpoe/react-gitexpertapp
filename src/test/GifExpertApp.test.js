import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas para <GifExpertApp />', () => {
  
  let wrapper  = shallow( <GifExpertApp  /> )

  test('01 Debe de renderizarse correctamente!', () => {
    console.log("renderizando  <GifExpertApp />")
    expect( wrapper ).toMatchSnapshot()
  })

  test('02 Debe de mostrar una lista de categorias!', () => {
    const categories = ["One Punch", "Superman"]    
    wrapper = shallow( <GifExpertApp defaultCategory = { categories } /> )

    expect(wrapper).toMatchSnapshot()
    expect( wrapper.find("GifGrid").length ).toBe( categories.length ) 

  });


})





