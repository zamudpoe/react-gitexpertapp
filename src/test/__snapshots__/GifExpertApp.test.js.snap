// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifExpertApp /> 01 Debe de renderizarse correctamente! 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__bounce animate__delay_2s"
  >
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Pruebas para <GifExpertApp /> 02 Debe de mostrar una lista de categorias! 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__bounce animate__delay_2s"
  >
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="0"
    />
    <GifGrid
      category="Superman"
      key="1"
    />
  </ol>
</Fragment>
`;
